subprojects/model-core/src/main/java/org/gradle/model/internal/reflect/problems/ExecutionValidationProblemId.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.model.internal.reflect.problems;
+
+import org.gradle.problems.ValidationProblemId;
+
+public enum ExecutionValidationProblemId implements ValidationProblemId {
```

Comment:
Splitting the original enum is one half of the problem. We also need to move the new enums with the more specific problem IDs to the subprojects that require them. That way we can remove the cyclic reference between `:model-core` and the subprojects like `:execution`.